OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -170,25 +191,25 @@ public void InflictDamage(Actor self, Actor attacker, Damage damage, bool ignore
 			{
 				// PERF: Util.ApplyPercentageModifiers has been manually inlined to
 				// avoid unnecessary loop enumerations and allocations
-				var appliedDamage = (decimal)damage.Value;
+				var appliedDamage = (float)damage.Value;
```

Comment:
Historically we don't use `float`s in simulation code to avoid desyncs.

Reply:
Yes, this will cause desyncs.

Reply:
I see. That explains the use of decimals.